File: 1470-shuffle-the-array/1470-shuffle-the-array.c


/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* shuffle(int* nums, int numsSize, int n, int* returnSize){
    int* answer = (int*)malloc(numsSize * sizeof(int));
    for (int i=0;i<n;i++) {
        answer[i*2] = nums[i];
        answer[(i+1)*2-1] = nums[i+n];
    }
    *returnSize = numsSize;
    return answer;
}

File: 1470-shuffle-the-array/1470-shuffle-the-array.py
class Solution:
    def shuffle(self, nums: List[int], n: int) -> List[int]:
        answer = [0] * (n*2)
        for i in range(0, n):
            answer[i*2] = nums[i]
            answer[(i+1)*2-1] = nums[i+n]
        return answer